# Circuit Theory - Jerzy Rutkowski
### *Chapter 2 - DC Analysis*

These are the so called drill problems found in Jerzy Rutkowski's online text book about circuit theory.

In [7]:
import numpy as np

## *Chapter 2.2* - Circuit Diagram and Kirchhof’s Laws

**Problem 1:**  

Calculate voltage $U_\mathrm{AB}$.

![P.2.2.1-diagram](P-2-2-1.png)

Via KVL,  

$$ 10\mathrm{V} - U_\mathrm{AC} - U_\mathrm{CD} = 0$$

and,
$$ U_\mathrm{BE} - U_\mathrm{GF} = 0$$

Via the above two equations and Ohm's Law,  

\begin{equation}\tag{1}
    10\mathrm{V} = 2\Omega\,I_\mathrm{AC} + 3\Omega\,I_\mathrm{CD}
\end{equation}

and,
\begin{equation}\tag{2}
    2\Omega\,I_\mathrm{BE} = 4\Omega\,I_\mathrm{GF}
\end{equation}    

Via the fact that there is a cutset around nodes $\mathrm{GF}$ in conjunction with KCL,  

$$ I_\mathrm{EF} - I_\mathrm{GB} = 0$$

Via the fact that $I_\mathrm{GB} = I_\mathrm{GE} = I_\mathrm{BE}$, the above equation, and substitution,  

\begin{equation}\tag{3}
    I_\mathrm{EF} - I_\mathrm{BE} = 0
\end{equation}    

Via KCL,  

$$ I_\mathrm{CE} + I_\mathrm{BE} - I_\mathrm{EF} = 0$$
$$ I_\mathrm{CE} + 0 = 0$$

\begin{equation}\tag{4}
    I_\mathrm{CE} = 0
\end{equation}    

Via KCL,  

\begin{equation}\tag{5}
    I_\mathrm{BE} + I_\mathrm{GF} = 9\mathrm{A}
\end{equation}    

Using equations (2) and (5) we can solve for the currents contained therein.

Via equation (2),  

$$ I_\mathrm{GF} = 0.5I_\mathrm{BE}$$

Via the above equation, equation (5), and substitution,  

$$ I_\mathrm{BE}\,(1 + 0.5) = 9\mathrm{A}$$

Thus,
\begin{equation}\tag{6}
    I_\mathrm{BE} = 6\mathrm{A}
\end{equation}    

Which means that,  

\begin{equation}\tag{7}
    U_\mathrm{BE} = 12\mathrm{V}
\end{equation}    

Now, to find $U_\mathrm{AC}$, we must first find $I_\mathrm{AC}$.  Via KCL,  

$$ I_\mathrm{AC} - I_\mathrm{CD} - I_\mathrm{CE} = 0$$
$$ I_\mathrm{AC} - I_\mathrm{CD} = 0$$

$$ I_\mathrm{AC} = I_\mathrm{CD}$$

Via the above equation, equation (1), and substitution,  

$$ 5\Omega\,I_\mathrm{AC} = 10\mathrm{V}$$

Dividing throughout by $5\Omega$ gives:  

\begin{equation}\tag{8}
    I_\mathrm{AC}= I_\mathrm{CD} = 2\mathrm{A}
\end{equation}    

Via the above equation and Ohm's Law,  

$$ U_\mathrm{AC} = 4\mathrm{V}$$

Via KVL,  

$$U_\mathrm{AC} + 7\mathrm{V} -U_\mathrm{AB} - U_\mathrm{BE} = 0$$

Thus,
\begin{align*}
    U_\mathrm{AB} &= U_\mathrm{AC} + 7\mathrm{V} - U_\mathrm{BE} \\
     &= (4 + 7 - 12)\,\mathrm{V} \\
     &= -1\mathrm{V}
\end{align*}     

**Problem 2:**  

![Fig-P.2.2.2](P-2-2-2.svg)

An ideal ammeter has zero resistance, implying zero voltage as well.  Assuming these conditions, via KVL,  

\begin{equation}\tag{2.1}
    U_\mathrm{AC} + U_\mathrm{BA} = 0
\end{equation}    

\begin{equation}\tag{2.2}
    U_\mathrm{DC} + U_\mathrm{BD} = 0
\end{equation}    

The number of branches is $b=5$ and the number of nodes is $n=4$, which means the number of *independent* KVL equations is $l=b-n+1=2$.  This implies that the above two equations are the only independent ones that can be obtained from the circuit.

The number of independent KCL equations is $t = n-1=3$.  Two of those are, via KCL,  

\begin{equation}\tag{2.3}
    I_\mathrm{AC} + I_\mathrm{DC} - I_\mathrm{CB} = 0
\end{equation}

\begin{equation}\tag{2.4}
    I_\mathrm{CB} - I_\mathrm{BA} - I_\mathrm{BD} = 0
\end{equation}    

Via the above and Ohm's Law,  

$$\frac{1}{8\Omega}\,U_\mathrm{AC} + \frac{1}{9\Omega}\,U_\mathrm{DC} - G_\mathrm{CB}U_\mathrm{CB} = 0$$

$$ G_\mathrm{CB}U_\mathrm{CB} - 0.5\mathrm{S}\,U_\mathrm{BA} - 1\mathrm{S}\,U_\mathrm{BD} = 0$$

Recall that '$G$' denotes the conductance of a circuit element, which is equal to the reciprocal of the resistance of that same element.  It's standard unit of measurement is the siemen ($\mathrm{S}$), which is, of course, equal to the reciprocal of one ohm ($1\Omega$).

Via the notion that the voltage running across an ideal ammeter is equal to zero, our assumption that the ammeter of our circuit is ideal, the above two equations, and substitution,  

\begin{equation}\tag{2.3a}
    \frac{1}{8\Omega}\,U_\mathrm{AC} + \frac{1}{9\Omega}\,U_\mathrm{DC} = 0
\end{equation}    

\begin{equation}\tag{2.4a}
    0.5\mathrm{S}\,U_\mathrm{BA} + 1\mathrm{S}\,U_\mathrm{BD} = 0
\end{equation}    

And, via Ohm's Law, the current flowing from node $\mathrm{A}$ to $\mathrm{C}$ is equal in magnitude, but opposite in direction, to the current flowing from node $\mathrm{D}$ to $\mathrm{C}$.  The same is true for the currents flowing though the top branches.  Thus, there is practically zero current flowing through our, assumed to be ideal, ammeter!

Via equations (2.1), (2.2), and (2.3a), in conjunction with substitution and multiplying throughout by $-8\Omega$,  

\begin{equation}\tag{2.3b}
    U_\mathrm{BA} + \frac{8}{9}\,U_\mathrm{BD} = 0
\end{equation}    

Via the equations (2.3b) and (2.4a) in conjunction with substitution,  

$$ 1\mathrm{S}\,U_\mathrm{BD} - \frac{4}{9\Omega}\,U_\mathrm{BD} = 0 $$

This didn't work, but we can use equations (2.1) up to and including (2.4) in the next method.

***Using another method:***  

The algebraic sum of the quantities in equations (2.3) and (2.4) is:  

\begin{equation}\tag{2.5}
    I_\mathrm{AC} + I_\mathrm{DC} = I_\mathrm{BA} + I_\mathrm{BD}
\end{equation}    

Note that the above equation is dependent on the aforementioned equations, neither of which can be invoked as a premise from this point forward.

So now we have three equations that are independent from one another, but four quantities to solve for.  We need one more equation, and it needs to contain a known quantity.  Thankfully, we are in luck!  Via KVL,  

\begin{equation}\tag{2.6}
    25\mathrm{V} + U_\mathrm{DC} - U_\mathrm{AC} = 0
\end{equation}    

Via equation (2.5) and Ohm's Law,  

\begin{equation}\tag{2.5a}
    \frac{1}{8\Omega}\,U_\mathrm{AC} + \frac{1}{9\Omega}\,U_\mathrm{DC} = 0.5\mathrm{S}\,U_\mathrm{BA} + 1\mathrm{S}\,U_\mathrm{BD}
\end{equation}    

Via the equations (2.5a), (2.1), and (2.2), in conjunction with substitution,  


$$\frac{1}{8\Omega}\,U_\mathrm{AC} + \frac{1}{9\Omega}\,U_\mathrm{DC} = - 0.5\mathrm{S}\,U_\mathrm{AC} - 1\mathrm{S}\,U_\mathrm{DC}$$

Multiplying throughout by $8\Omega$ gives:  

$$U_\mathrm{AC} + \frac{8}{9}\,U_\mathrm{DC} = -4U_\mathrm{AC} - 8U_\mathrm{DC}$$

$$ 37\,U_\mathrm{AC} = - 80U_\mathrm{DC}$$

\begin{equation}\tag{2.7}
    U_\mathrm{DC} = -\frac{37}{80}\,U_\mathrm{AC}
\end{equation}    

Via equations (2.6) and (2.7) in conjunction with substitution,  

$$ 25\mathrm{V} = U_\mathrm{AC} + \frac{37}{80}\,U_\mathrm{AC}$$

$$ 25\mathrm{V} = \frac{117}{80}\,U_\mathrm{AC}$$

\begin{equation}\tag{2.8}
    U_\mathrm{AC} = \frac{80 \cdot 25}{117}\,\mathrm{V}\approxeq 17.09\mathrm{V}
\end{equation}    

Via equations (2.8) and (2.7), in conjunction with substitution,  

\begin{equation}\tag{2.7a}
    U_\mathrm{DC} = - \frac{37\cdot25}{117}\,\mathrm{V} \approxeq -7.91\mathrm{V}
\end{equation}    

And, via equations (2.1), (2.2), (2.8), and (2.7a), in conjunction with equivalence principle,  

\begin{equation}\tag{2.1a}
    U_\mathrm{BA} = -U_\mathrm{AC} \approxeq -17.09\mathrm{V}
\end{equation}    

and,
\begin{equation}\tag{2.2a}
    U_\mathrm{BD} = -U_\mathrm{DC} \approxeq 7.91\mathrm{V}
\end{equation}    

The currents are, via the above four equations and Ohm's Law,  

\begin{align*}
    I_\mathrm{AC} =\frac{250}{117}\,\mathrm{A} &\approxeq 2.14\mathrm{A}\\ \\
    I_\mathrm{DC} &\approxeq - 878.44\mathrm{mA} \\ \\
    I_\mathrm{BA} = -\frac{40\cdot25}{117}\,\mathrm{A} &\approxeq -8.55\mathrm{A} \\ \\
    I_\mathrm{BD} &\approxeq 7.91\mathrm{A}
\end{align*}    

Note that it also follows from KVL that $25\mathrm{V} + U_\mathrm{BA} - U_\mathrm{BD} = 0$, however this also follows from equations (2.1), (2.2), and (2.6), in conjunction with some other basic principles of math.  Thus, that equation, together with the other three equations, are not fully co-independent.  If this equation were use with the other three, our attempt at solving for the quantities contained therein would fail.  Below is the proof.

Via equations (2.1) and (2.2) in conjunction with equivalence principle -- or, to put it another way, the algebraic sum of those equations is --  

$$ U_\mathrm{AC} + U_\mathrm{BA} = U_\mathrm{DC} + U_\mathrm{BD} $$

Shuffling some quantities around in the above equations gives:  

$$ U_\mathrm{AC} - U_\mathrm{DC} = U_\mathrm{BD} - U_\mathrm{BA}$$

Multiplying throughout by negative one gives:  

$$ U_\mathrm{DC} - U_\mathrm{AC} = U_\mathrm{BA} - U_\mathrm{BD}$$

Via the above equation, equation (2.6), and substitution,  

$$ 25\mathrm{V}+U_\mathrm{BA}-U_\mathrm{BD} = 0$$

Q.E.D.

***Using another method:***

Via KVL,  

\begin{equation}\tag{2.1}
    U_\mathrm{AC} + U_\mathrm{BA} + U_\mathrm{CB} = 0
\end{equation}    

and,
\begin{equation}\tag{2.2}
    U_\mathrm{CB} - U_\mathrm{DC} - U_\mathrm{BD} = 0
\end{equation}    

Via the above two equations and substitution,  

\begin{equation}\tag{2.3}
    U_\mathrm{DC} + U_\mathrm{BD} + U_\mathrm{AC} + U_\mathrm{BA} = 0
\end{equation}    

Via the above equation and Ohm's Law,  

\begin{equation}\tag{2.4}
    9\Omega\,I_\mathrm{DC} + 1\Omega\,I_\mathrm{BD} + 8\Omega\,I_\mathrm{AC} + 2\Omega\,I_\mathrm{BA} = 0
\end{equation}    

Via KCL,  

$$ I_\mathrm{AC} + I_\mathrm{DC} - I_\mathrm{CB} = 0$$

and,
$$ I_\mathrm{CB} - I_\mathrm{BA} - I_\mathrm{BD} = 0$$

Via the above two equations and substitution,  

\begin{equation}\tag{2.5}
    I_\mathrm{AC} + I_\mathrm{DC} - I_\mathrm{BA} - I_\mathrm{BD} = 0
\end{equation}    

Via KVL,  

$$ 25\mathrm{V} - U_\mathrm{AC} + U_\mathrm{DC} = 0$$

and,
$$ 25\mathrm{V} + U_\mathrm{BA} - U_\mathrm{BD} = 0$$

Via the above two equations and Ohm's Law,  

$$ 25\mathrm{V} - 8\Omega\,I_\mathrm{AC} + 9\Omega\,I_\mathrm{DC} = 0$$

and,
$$ 25\mathrm{V} + 2\Omega\,I_\mathrm{BA} - 1\Omega\,I_\mathrm{BD} = 0$$

Which leaves us with:  

\begin{equation}\tag{2.6}
    I_\mathrm{BD} = 25\mathrm{A} + 2I_\mathrm{BA}
\end{equation}    

and,
\begin{equation}\tag{2.7}
    I_\mathrm{AC} = \frac{25}{8}\,\mathrm{A} + \frac{9}{8}\,I_\mathrm{DC}
\end{equation}    

Via the above two equations, equation (2.5), and substitution,  

$$\frac{25}{8}\,\mathrm{A} + \frac{9}{8}\,I_\mathrm{DC} + I_\mathrm{DC}  - 25\mathrm{A} - 3I_\mathrm{BA} = 0$$

\begin{equation}\tag{2.8}
    \frac{25}{8}\,\mathrm{A} - 25\mathrm{A} - 3I_\mathrm{BA} + \frac{17}{8}\,I_\mathrm{DC} =0
\end{equation}    

Multiplying throughout by $8$ gives:  

\begin{equation}\tag{2.8}
    17I_\mathrm{DC} - 24I_\mathrm{BA} -175\mathrm{A} = 0
\end{equation}    

Via equations (2.4), (2.6), and (2.7), in conjunction with substitution,  

$$ 9\Omega\,I_\mathrm{DC} + 25\mathrm{V} + 4\Omega\,I_\mathrm{BA} + 25\mathrm{V} + 9\Omega\,I_\mathrm{DC} = 0 $$

Thus,
\begin{equation}\tag{2.9}
    18\Omega\,I_\mathrm{DC} + 50\mathrm{V} + 4\Omega\,I_\mathrm{BA} = 0
\end{equation}    

Dividing throughout equation (2.8) by $6$ gives:  

$$ 4I_\mathrm{BA} = \frac{17}{6}\,I_\mathrm{DC} - \frac{175}{6}\,\mathrm{A} $$

Via the above equation, equation (2.9), and substitution,  

$$ 18\Omega\,I_\mathrm{DC} + 50\mathrm{V} + \frac{17}{6}\,\Omega\,I_\mathrm{DC} - \frac{175}{6}\,\mathrm{V} = 0$$

Multiplying throughout by $6$ gives:  

$$ 108\Omega\,I_\mathrm{DC} + 300\mathrm{V} + 17\Omega\,I_\mathrm{DC} - 175\mathrm{V} = 0$$

$$ 125\Omega\,I_\mathrm{DC} + 125\mathrm{V} = 0  $$

Thus,
\begin{equation}\tag{2.10}
    I_\mathrm{DC} = -1\mathrm{A}
\end{equation}    

Via the above equation, equation (2.9), and substitution,  

$$ 50\mathrm{V} - 18\mathrm{V} + 4\Omega\,I_\mathrm{BA} = 0$$

$$ 4\Omega\,I_\mathrm{BA} = -32\mathrm{V}$$

Dividing throughout by $4\Omega$ gives:

\begin{equation}\tag{2.11}
    I_\mathrm{BA} = -8\mathrm{A}
\end{equation}    

Via the above equation, equation (2.6), and substitution,  

\begin{equation}\tag{2.12}
    I_\mathrm{BD} = 25\mathrm{A} - 8\mathrm{A} = 17\mathrm{A}
\end{equation}

Via equation (2.5),  

\begin{align*}\tag{2.13}
    I_\mathrm{AC} &= I_\mathrm{BD} + I_\mathrm{BA} - I_\mathrm{DC} \\
     &= 17\mathrm{A} - 8\mathrm{A} + 1\mathrm{A} \\
     &= 10\mathrm{A}
\end{align*}     

Note that the negative values for $I_\mathrm{BA}$ and $I_\mathrm{DC}$ imply that the assumption about the direction of the conventional current flow that is made by the corresponding arrowheads in [Figure P.2.2.2](P-2-2-2.svg) is incorrect.  Below is a revised version of said figure with arrowheads that point in the direction of the actual conventional flow.  

![Fig-2.2.2-revised-arrowheads](P-2-2-2-rev-arrowheads.svg)

Now, using the arrowheads in the revised (above) figure, the voltage running across each resistor is, via the corresponding current values found above, Ohm's Law, and substitution,  

$$ U_\mathrm{AC} = 80\mathrm{V}$$

$$ U_\mathrm{AB} = 16\mathrm{V}$$

$$ U_\mathrm{BD} = 17\mathrm{V}$$

$$ U_\mathrm{CD} = 9\mathrm{V}$$

_This method didn't work!_

**Problem 3:**  

What should the value of $R$ be so that the current is $I=0.5\mathrm{A}$?

Assuming $I$ is the current flowing through the voltage source as opposed to the right hand branch of the parallel circuit,

$$ U_1 = 4\mathrm{V} = R\cdot I_1$$

In [2]:
U1 = 5-1

$$ U_1 = U_2 = 6\Omega \, I_2 =4\mathrm{V}$$

$$ I_2 = \frac{2}{3}\,\mathrm{A}$$

Via KCL,  

$$ I = 0.5\mathrm{A}= I_1 + I_2$$

Thus,
$$ I_1 = 0.5\mathrm{A} - \frac{2}{3}\,\mathrm{A} = \frac{1}{2}\,\mathrm{A} - \frac{2}{3}\,\mathrm{A} = -\frac{1}{6}\,\mathrm{A}$$

Via the first equation of this problem's attempt at a solution,  

$$ R = \frac{4\mathrm{V}}{I_1} = -24\Omega$$

In [3]:
0.5*(2 +1/(1/6-1/24))

5.0

Assuming $I$ is the current flowing through the right hand branch,  

Via Ohm's Law,  

$$U = 6\Omega\cdot0.5\mathrm{A}= 3\mathrm{V} $$

Via KCL,  

$$ I_2 = I_1 + I = 0.5\mathrm{A} + I_1$$

Via KVL,  

$$ U_2 = 5\mathrm{V} - U = 2\mathrm{V}$$

Via the above equation and Ohm's Law,  

$$ I_2 = \frac{2\mathrm{V}}{2\Omega} = 1\mathrm{A}$$

Via KCL and the above equation,  

$$ I_1 = 1\mathrm{A} - 0.5\mathrm{A} = 0.5\mathrm{A}$$

Via the above equation, Ohm's Law, and KVL,  

$$ RI_1 = R\cdot0.5\mathrm{A} = 3\mathrm{V}$$

$$ R = 6\Omega$$

**Problem 6:**  

Suppose the indicated voltage, denoted as '$U_\mathrm{M}$', is $12\mathrm{V}$.  Find $R$.

The voltage running across the top left resistor, denoted as '$U_\mathrm{TL}$', is given by:  

$$ U_\mathrm{TL} = 4\Omega\cdot4\mathrm{A} = 16\mathrm{V}$$

Via the above equation, KVL, and the presupposition that $U_\mathrm{M}=12\mathrm{V}$, the supply voltage ($E$) is:

$$ E - 16\mathrm{V} - 12\mathrm{V} = 0$$

$$ E = 28\mathrm{V}$$

Via the above equation and Ohm's Law,  

$$ U_\mathrm{M} = R\cdot I_\mathrm{M} = 12\mathrm{V}$$

Via KCL,  

$$ I := I_\mathrm{L} = 4\mathrm{A} = I_\mathrm{M} + I_\mathrm{R}$$

Via KVL and Ohm's Law,  

$$ U_\mathrm{M} - U_\mathrm{R} = 0$$

$$ 12\mathrm{V} = U_\mathrm{R} = (1\Omega + 3\Omega)\,I_\mathrm{R} = 4\Omega\,I_\mathrm{R}$$

Dividing throughout by $4\Omega$ gives:  

$$ I_\mathrm{R} = 3\mathrm{A}$$

Via the above equation, KVL, and substitution,  

$$ I_\mathrm{M} = 1\mathrm{A}$$

Finally,  

$$ R = \frac{12\mathrm{V}}{1\mathrm{A}} = 12\Omega$$

**Problem 7:**  

Find the power absorbed (or supplied?) by the current source.

$$ U_\mathrm{AD}= 7\Omega\,I_\mathrm{AD} = 5\mathrm{V}$$

$$I_\mathrm{AD} = \frac{5\mathrm{V}}{7\Omega} \approxeq 714.29\mathrm{mA} $$

Via KCL,  

$$ I_\mathrm{DA} - I_\mathrm{AD} - I_\mathrm{AB} \approxeq I_\mathrm{DA} - 2.714\mathrm{A} \approxeq 0$$

$$ I_\mathrm{DA} \approxeq 2.71\mathrm{A}$$

Via KVL and Ohm's Law,  

$$ 6\Omega\,I_{\mathrm{BC}1} = 4\Omega\,I_{\mathrm{BC}2}$$

Via KCL,  

$$ I_{\mathrm{BC}1} + I_{\mathrm{BC}2} = 2\mathrm{A}$$

$$ I_{\mathrm{BC}2} = \frac{3}{2}\,I_{\mathrm{BC}1}$$

$$ I_{\mathrm{BC}1}\,\left(1 + \frac{3}{2} \right) = \frac{5}{2}\,I_{\mathrm{BC}1} = 2\mathrm{A}$$

Thus,  

$$ I_{\mathrm{BC}1} = \frac{4}{5}\,\mathrm{A}$$

Via the above equation and Ohm's Law,  

$$ U_\mathrm{BC} = 6\Omega\cdot \frac{4}{5}\,\mathrm{A} = 4.8\mathrm{V}$$

Via KVL,  

$$ 5\mathrm{V} + U_\mathrm{AB} - 4.8\mathrm{V} = U_\mathrm{AB} + 0.2\mathrm{V} =0$$

$$ U_\mathrm{AB} = -0.2\mathrm{V}$$

Thus,  

$$ P = -0.2\mathrm{V}\cdot2\mathrm{A} = -0.4\mathrm{W}$$

And, since the current source's arrowheads follow active sign convention in conjunction with the fact that $P$ has a negative value, the current source is *absorbing* $0.4\mathrm{W}$ of power.

**Problem 8:**  

Find the supply voltage ($E$) of a ladder network shown, so that $I=2\mathrm{A}$.  Assume that $R=5\Omega$.

Via KVL and Ohm's Law,  

$$ E = 0.5RI_\mathrm{AB} + RI_\mathrm{BG}$$

Note that, via the presupposition of this problem that $I=2\mathrm{A}$:  

$$ I = I_\mathrm{CD} = I_\mathrm{DE} = I_\mathrm{EF} = 2\mathrm{A}$$

Via the above and Ohm's Law,  

$$ U_\mathrm{CD} = U_\mathrm{DE} = \frac{5}{2}\,\Omega\cdot2\mathrm{A} = 5\mathrm{V}$$

Via the above, Ohm's Law, and KVL,  

$$ U_\mathrm{CF} = 5\Omega\,I_\mathrm{CF} = 10\mathrm{V}$$

Dividing throughout by $5\Omega$ gives:  

$$ I_\mathrm{CF} = 2\mathrm{A}$$

Via KCL,  

$$ I_\mathrm{BC} = 2\mathrm{A} + 2\mathrm{A} = 4\mathrm{A}$$

Via the above and Ohm's Law,  

$$ U_\mathrm{BC} = 10\mathrm{V}$$

Via Ohm's Law and KVL,  

$$ U_\mathrm{BG} = 5\Omega\,I_\mathrm{BG} = 10\mathrm{V} + 10\mathrm{V} = 20\mathrm{V}$$

Dividing throughout by $R$ gives:  

$$ I_\mathrm{BG} = 4\mathrm{A}$$

Via KCL,  

$$ I_\mathrm{AB} = 4\mathrm{A} + 4\mathrm{A} = 8\mathrm{A}$$

Via the above and Ohm's Law,  

$$ U_\mathrm{AB} = 20\mathrm{V}$$

Finally, via KVL,  

$$ E = 20\mathrm{V} + 20\mathrm{V} = 40\mathrm{V}$$

**Problem 9:**  

Find $R$ if $I=2\mathrm{A}$.  Then, find all currents and voltages.

The middle voltage drop is:  

$$ U_\mathrm{m} = 4\Omega\cdot2\mathrm{A} = 8\mathrm{V}$$

Via KVL and Ohm's Law,  

$$ U_\mathrm{r} = RI_\mathrm{r} = 8\mathrm{V}$$

Via KCL,  

$$ 6\mathrm{A} = 2\mathrm{A} + I_\mathrm{r}$$

$$ I_\mathrm{r} = 4\mathrm{A}$$

Via the above three equations,  

$$ R = \frac{8\mathrm{V}}{4\mathrm{A}} = 2\Omega$$

And, now for the remaining unkown voltages.

Via Ohm's Law, the voltage dropped across the left hand resistor is:  

$$ U_\mathrm{l} = 2\Omega\cdot6\mathrm{A} = 12\mathrm{V}$$

And, via KVL, the voltage running across the current source, denoted as '$E$', is:  

$$ E = U_\mathrm{l} + U_\mathrm{m} = 12\mathrm{V} + 8\mathrm{V} = 20\mathrm{V}$$

Note that the current source's arrowheads satisfy active sign conventions.

**Problem 10:**  

By what factor should the $6\mathrm{A}$ current source in the above problem's circuit be increased to double the power it supplies ($P$)?  By what factor should it be increased to double the current $I=I_\mathrm{m}$?

First recall that,  

$$ P = I_\mathrm{l}E$$

and,
$$ E = U_\mathrm{l} + U_\mathrm{m} = 2\Omega\,I_\mathrm{l} + 4\Omega\,I_\mathrm{m}$$

It might seem obvious at first that we, based on the first equation of our attempt at a solution, should simply double the current to double the power supplied.  But, as we can see with the second equation, the voltage running across the current source, denoted as '$E$', is dependent on the current source's output current.  Once we unpack the first equation, we see that the relationship between the supplied power and the output current is not linear, as it seemed at first.

What's more,  $I_\mathrm{m}$ is also dependent on the output current.  Via KCL,  

$$ I_\mathrm{l} = I_\mathrm{m} + I = I_\mathrm{m} + 2\mathrm{A}$$

Thus,
$$ I_\mathrm{m} = I_\mathrm{l} - 2\mathrm{A}$$

and,
\begin{align*}
    E &= 2\Omega\,I_\mathrm{l} + 4\Omega\,(I_\mathrm{l} - 2\mathrm{A}) \\
     &= 6\Omega\,I_\mathrm{l} - 8\mathrm{V}
\end{align*}     

The supplied power as a function of the output current is thus,  

$$ P := P(I_\mathrm{l}) = 6\Omega\,I_\mathrm{l}^2 - 8\mathrm{V}\,I_\mathrm{l}$$

To find our factor, we will first solve for $x:=\grave{I_\mathrm{l}}$, where $x$ is equal to the product of $I_\mathrm{l}$ and the factor we are looking for, denoted as '$\alpha$', in the quadratic equation below:  

$$ 6\Omega\,x^2 - 8\mathrm{V}\,x = 2P = 240\mathrm{W}$$

$$ 6\Omega\,x^2 - 8\mathrm{V}\,x - 240\mathrm{W} = 0$$

In [10]:
# Implementation of the Quadratic Formula
a = 6
b = -8
c = -240
x = set([(-b+np.sqrt(b**2-4*a*c))/(2*a), (-b-np.sqrt(b**2-4*a*c))/(2*a)])
x

{-5.692928009446304, 7.026261342779637}

Going with the positive part of the solution for $x$, we have:  

\$$ \alpha I_\mathrm{l} \approxeq 7.03\mathrm{A}$$

$$ \alpha \approxeq 1.17$$

In [12]:
[x_i/6 for x_i in x]

[-0.9488213349077174, 1.1710435571299396]

![Fig - 2.2.9](P-2-2-9.png)

Now, for the second part of the question, first let $\grave{I}$ be defined as $\grave{I}:=2I=4\mathrm{A}$ and $x$ and $\grave{I_\mathrm{l}}$ be defined as $x:=\grave{I_\mathrm{l}}:=\alpha I_\mathrm{l}$, where $alpha$ is the factor we are looking for.  

Next we need to find the relationship between the output current ($x=\grave{I_\mathrm{l}}$) and the middle current ($\grave{I}=\grave{I_\mathrm{m}}$).

Via KCL,  

$$ \grave{I_\mathrm{l}} - \grave{I_\mathrm{m}} - \grave{I_\mathrm{r}} = 0$$

The relationship between the right and middle currents is, via Ohm's Law and KVL,  

$$ 4\Omega\,\grave{I_\mathrm{m}} = R\,\grave{I_\mathrm{r}} = 2\Omega\,\grave{I_\mathrm{r}}$$

Via the above two equations and equivalence principle,  

$$ \grave{I_\mathrm{l}} - \grave{I_\mathrm{m}} - \frac{4\Omega}{R}\,\grave{I_\mathrm{m}} = \grave{I_\mathrm{l}} - 3\grave{I_\mathrm{m}} =0$$

Thus,
$$ x = 3\grave{I} = 12\mathrm{A}$$

Finally,  

$$ x = \alpha I_\mathrm{l} = \alpha 6\mathrm{A} = 12\mathrm{A}$$

$$ \alpha = 2$$